### PR TITLE
document activate --temp

### LINKED ISSUE #1944: document --temp
Could not find any documentation online for the `--temp` flag for `activate`. It also does not show for `?Pkg.activate`, only for `?activate` (from `pkg` mode).

```
julia> using Pkg

help?> Pkg.activate
  Pkg.activate([s::String]; shared::Bool=false)

  Activate the environment at s. The active environment is the environment that is modified by executing package commands. The logic for what path is activated is as follows:

    •    If shared is true, the first existing environment named s from the depots in the depot stack will be activated. If no such environment exists, create and activate that environment in the first depot.

    •    If s is an existing path, then activate the environment at that path.

    •    If s is a package in the current project and s is tracking a path, then activate the environment at the tracked path.

    •    Otherwise, s is interpreted as a non-existing path, which is then activated.

  If no argument is given to activate, then activate the home project. The home project is specified by either the --project command line option to the julia executable, or the JULIA_PROJECT environment variable.

  Examples
  ≡≡≡≡≡≡≡≡≡≡

  Pkg.activate()
  Pkg.activate("local/path")
  Pkg.activate("MyDependency")
```

```
(@v1.5) pkg> ?activate
  activate
  activate [--shared|--temp] [path]

  Activate the environment at the given path, or the home project environment if no path is specified. The active environment is the environment that is modified by executing package commands. When the option --shared is given,
  path will be assumed to be a directory name and searched for in the environments folders of the depots in the depot stack. In case no such environment exists in any of the depots, it will be placed in the first depot of the
  stack. Use the temp option to create temporary environments. This should be useful for experimenting with packages.
```

### PR DESCRIPTION
**Pull request by oxinabox**
closes #1944

### CODE REVIEW

File: src/Pkg.jl [hunk 1]
```diff
@@ -400,6 +400,9 @@ The logic for what path is activated is as follows:
   * If `shared` is `true`, the first existing environment named `s` from the depots
     in the depot stack will be activated. If no such environment exists,
     create and activate that environment in the first depot.
+  * If `temp` is `true` this will create and activate a temporary enviroment which will
+    be deleted when the julia process is exitted. The `s` argument must not be passed
```
Comment (oxinabox) [inline src/Pkg.jl:404]:
```suggestion
    be deleted when the julia process is exited. The `s` argument must not be passed
```

File: src/Pkg.jl [hunk 2]
```diff
@@ -391,7 +391,7 @@ const status = API.status
 
 
 """
-    Pkg.activate([s::String]; shared::Bool=false)
+    Pkg.activate([s::String]; temp::Bool=false, shared::Bool=false)
```
Comment (fredrikekre) [inline src/Pkg.jl:394]:
```suggestion
    Pkg.activate([s::String]; shared::Bool=false)
    Pkg.activate(; temp::Bool=false)
```

File: src/Pkg.jl [hunk 3]
```diff
@@ -400,6 +400,9 @@ The logic for what path is activated is as follows:
   * If `shared` is `true`, the first existing environment named `s` from the depots
     in the depot stack will be activated. If no such environment exists,
     create and activate that environment in the first depot.
+  * If `temp` is `true` this will create and activate a temporary enviroment which will
+    be deleted when the julia process is exited. The `s` argument must not be passed
+    with the `temp` option set to `true`.
```
Comment (fredrikekre) [inline src/Pkg.jl:405]:
```suggestion
    be deleted when the julia process is exited.
```

File: src/REPLMode/command_declarations.jl [hunk 4]
```diff
@@ -256,7 +256,8 @@ The active environment is the environment that is modified by executing package
 When the option `--shared` is given, `path` will be assumed to be a directory name and searched for in the
 `environments` folders of the depots in the depot stack. In case no such environment exists in any of the depots,
 it will be placed in the first depot of the stack.
-Use the `temp` option to create temporary environments. This should be useful for experimenting with packages.
+Use the `--temp` option to create temporary environments which are removed when the julia
+process is exited. This is useful for experimenting with packages.
```
Comment (fredrikekre) [inline src/REPLMode/command_declarations.jl:260]: Don't think we need to motivate this?
```suggestion
process is exited.
```

File: src/Pkg.jl (hunk 2 above)
Comment (oxinabox) [reply src/Pkg.jl:394]: Oh that is much better

File: src/REPLMode/command_declarations.jl (hunk 4 above)
Comment (oxinabox) [reply src/REPLMode/command_declarations.jl:260]: I don't think so either, but i wasn't sure i wanted to fully delete the motivation that was alread there

**Pull request merged**